src/sql/src/func.rs

```diff
@@ -1792,7 +1792,7 @@ lazy_static! {
                 params!(Oid, Int32) => sql_impl_func(
                     "CASE
                         WHEN $1 IS NULL THEN NULL
-                        ELSE coalesce((SELECT concat(name, mz_internal.mz_render_typemod($1, $2)) FROM mz_catalog.mz_types WHERE oid = $1), '???')
+                        ELSE coalesce((SELECT concat(coalesce(mz_internal.mz_type_name($1), name), mz_internal.mz_render_typemod($1, $2)) FROM mz_catalog.mz_types WHERE oid = $1), '???')
```

Comment:
The `coalesce` here ensures we still handle user-defined types (or types that are not in the pgrepr pattern match) properly. An alternative implementation would have been to add a new column to `mz_types` instead, but I thought having an internal function was better than adding more stuff to an internal table.